mne/encoding/feature.py

```diff
@@ -0,0 +1,250 @@
+"""A collection of classes for transforming, preprocessing, and fitting."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Transformer for creating delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    time_window : array, shape (2,)
+        Specifies a minimum/maximum delay in seconds. In this case, all
+        delays between `min(time_window)` and `max(time_window)`
+        will be calculated using `sfreq`.
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
```

Comment:
as in don't support time windows? I can remove that code though it is
pretty simple to include (basically just a matter of calculating `period =
1. / sfreq` then doing `for tmin, tmax in windows:
delays.append(np.arange(tmin, tmax + period, period)`

On Tue, Aug 2, 2016 at 10:45 AM, Jean-Rémi KING <notifications@github.com>
wrote:

> In mne/encoding/feature.py
> <https://github.com/mne-tools/mne-python/pull/3310#discussion_r73202602>:
>
> > +from ..utils import warn
> > +
> > +
> > +class FeatureDelayer(object):
> > +    """Transformer for creating delayed versions of an input stimulus.
> > +
> > +    Parameters
> > +    ----------
> > +    time_window : array, shape (2,)
> > +        Specifies a minimum/maximum delay in seconds. In this case, all
> > +        delays between `min(time_window)` and `max(time_window)`
> > +        will be calculated using `sfreq`.
> > +    delays: array of floats, shape (n_delays,)
> > +        The time (in seconds) of each delay for specifying
> > +        pre-defined delays. Negative means timepoints in the past,
> > +        positive means timepoints in the future.
>
> For now, I would only support an explicit list/slice. Let's do the higher
> level API with floats in a subsequent PR to try to simplify the review of
> this PR
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3310/files/15a5bb641b7f7a9edf70f48f531590d992a366ab#r73202602>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABwSHQLwh5fuFZwSItiQitTFmXBfTAEvks5qb4I1gaJpZM4I28Ut>
> .
>